File: src/main/java/com/asos/reservationSystem/domain/dto/CourseDto.java
package com.asos.reservationSystem.domain.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CourseDto {
    private Long id;
    private String name;
    private Double price;
    private String room;
    private UserDto teacher;
    private List<UserDto> students;
}

File: ASOS-Frontend/DTOs/CourseDto.cs
using System.Text.Json.Serialization;
using DTOs.Account;

namespace DTOs
{
    public class CourseDto
	{
		[JsonPropertyName("id")]
		public long? Id { get; set; }

		[JsonPropertyName("name")]
		public string Name { get; set; }

		[JsonPropertyName("price")]
		public double Price { get; set; }

		[JsonPropertyName("room")]
		public string Room { get; set; }

		[JsonPropertyName("teacher")]
		public UserDto? Teacher { get; set; }

		[JsonPropertyName("students")]
		public List<UserDto>? Students { get; set; }
	}
}
